src/java_bytecode/java_types.cpp

```diff
@@ -273,7 +273,7 @@ std::string gather_full_class_name(const std::string &src)
 
   class_name = erase_type_arguments(class_name);
 
-  std::replace(class_name.begin(), class_name.end(), '.', '$');
+  std::replace(class_name.begin(), class_name.end(), '$', '.');
```

Comment:
I need to verify this more carefully - but the reason for it being this way round is to get inner inner classes being represented in a specific manner. The example in your commit message is _not_ a valid java bytecode type descriptor by my reading of [4.2.1](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.2.1)